rust/worker/src/blockstore/arrow/blockfile.rs

```diff
@@ -167,7 +167,20 @@ impl ArrowBlockfileWriter {
         } else {
             let (split_key, new_delta) = delta.split::<K, V>();
             self.sparse_index.add_block(split_key, new_delta.id);
-            new_delta.add(prefix, key, value);
+            match new_delta.get_min_key() {
+                Some(min_key) => {
```

Comment:
I don't know if this handles all cases, what if the split + add into the new delta throws the block over the max size again? Imagine a case with a block with three values (1,3,5), and the split leave (1,3), (5). We are adding 2 but (1,2,3) would need to be split. I think we need to replay the logic, this regression occurred in #2052 (a terrible PR I authored). It would recurse the add

Reply:
I thought about this case. Won't the recursion depth be at most 1? IMO we should be ok to overflow by one key-value pair in favor of simplicity. we should not complicate this logic. If we try to add further keys to (1,2,3) it would split automatically in that insertion round.

Reply:
Yes it will be at most one. I don't view this as overcomplicating the logic. We shouldn't allow the max block size to be exceeded. Also this actually simplifies the splitting logic by localizing it to one area.